File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int n = s.size();
        int idx = 0;
        for(int i=0;i<t.size();i++){
            if(s[idx]==t[i]) 
                idx++;
        }
        return (idx==n);
    }
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) == 0:
            return True
        idx = 0
        for ch in t:
            if idx < len(s):
                if ch == s[idx]:
                    idx += 1
            else:
                break
        return idx == len(s)